CHANGELOG/changelog-md-template.lua

```diff
@@ -0,0 +1,95 @@
+return [[
+# ${version}
+
+> local function render_changelog_entry(entry)
+- ${entry.message}
+>   if entry.pr then
+  [#${entry.pr.id}](${entry.pr.link})
+>   end
+>   if entry.jiras then
+>     for _, jira in ipairs(entry.jiras or {}) do
+  [${jira.id}](${jira.link})
+>     end
+>   end
+>   if #(entry.issues or {}) > 0 then
+(issue:
+>     for _, issue in ipairs(entry.issues or {}) do
+ [#${issue.id}](${issue.link})
+>     end
+)
+>   end
+> end
+
+
+> local function render_changelog_entries(entries)
+>   for _, entry in ipairs(entries or {}) do
+>     render_changelog_entry(entry)
+>   end
+> end
+
+
+> local function render_changelog_section(section_name, t)
+>   if not is_empty_table(t) then
+### ${section_name}
+
+>   end
+>   for scope, entries in pairs(t or {}) do
+>     if scope ~= "" then
+#### ${scope}
+
+>     end
+>     render_changelog_entries(entries)
+>   end
+> end
```

Comment:
It would be good to have ordering of individual entries follow some sort of consistent pattern. The ordering is deliberate in the sections below (breaking changes, deprecations, dependencies, etc...), so individual items _within_ a section should also follow some sort of consistent pattern.

For `scopes` we may want to hard-code the order by some sort of subjective level of significance. Example: core, then plugins, then pdk, and so on. _Or_ we could just `table.sort` the keys alphanumerically.

For entries _within_ a scope, maybe we should sort according to the PR # so that they appear at least _somewhat_ chronologically.

Reply:
Other benefit of explicit sorting is that the same input always produces the same output .md file no matter which platform/filesystem/Lua version.

Reply:
LGTM.

The scope is now prioritized.
```
local SCOPE_PRIORITY = { -- smallest on top
  Performance = 10,
  Configuration = 20,
  Core = 30,
  PDK = 40,
  Plugin = 50,
  ["Admin API"] = 60,
  Clustering = 70,
  Default = 100, -- default priority
}
setmetatable(SCOPE_PRIORITY, {
  __index = function()
    return rawget(SCOPE_PRIORITY, "Default") - 1
  end
})
```

Files now added  explicit sorting to avoid the issue that you mentioned.

Regarding the section. I think the section order can be edit as it's now hardcoded.
```
> render_changelog_section("Breaking Changes", enterprise.breaking_changes)
> render_changelog_section("Deprecations", enterprise.deprecations)
> render_changelog_section("Dependencies", enterprise.dependencies)
> render_changelog_section("Features", enterprise.features)
> render_changelog_section("Fixes", enterprise.bugfixes)
```